enrichment-definition/src/main/java/eu/europeana/enrichment/model/impl/TopicEntityImpl.java

```diff
@@ -0,0 +1,219 @@
+
+@Entity(value="TopicEntityImpl")
+public class TopicEntityImpl implements TopicEntity {
+	
+	//id will be used for storing MongoDB _id
+	@Id
+	@JsonIgnore
+	public String _id = new ObjectId().toString();
+	
+	@Transient
+	private TopicModel model;
+	
+	// <{baseurl}/topic/1>
+	private String topicID;
+	
+	// "LDA_EXP1-K15-IT200#TOPIC1"
+	private String identifier;
+	
+	private List<String> labels;
+	// language based description
+	private Map<String,String> descriptions;
+	
+	
+	private List<TermEntity> terms;
+	
+	private List<TermEntity> keywords;
+	
+	private String modelId;
+	
+	private Date created;
+	private Date modified;
+	
+	
+	public TopicEntityImpl()
+	{
+		
+	}
+
+	
+
+	public TopicEntityImpl(TopicEntity topicEntity) {
+		this.topicID = topicEntity.getTopicID();
+		this.identifier = topicEntity.getIdentifier();
+		this.labels = topicEntity.getLabel();
+		this.descriptions = topicEntity.getDescription();
+		this.terms = topicEntity.getTopicTerms();
+		this.keywords = topicEntity.getTopicKeywords();
+		this.model = topicEntity.getTopicModel();
+		this.modelId = topicEntity.getModelId();
+		this.created = topicEntity.getCreatedDate();
+		if (topicEntity.getModifiedDate()==null)
+			this.modified = new Date();
+		else
+			this.modified = topicEntity.getModifiedDate();
+	}
+
+	@Override
+	public String getTopicID() {
+		return this.topicID;
+	}
+
+	@Override
+	public void setTopicID(String id) {
+		this.topicID = id;
+
+	}
+
+	@Override
+	public String getIdentifier() {
+		return this.identifier;
+	}
+
+	@Override
+	public void setIdentifier(String identifier) {
+		this.identifier = identifier;
+
+	}
+
+	@Override
+	public List<String> getLabel() {
+		return this.labels;
+	}
+
+	@Override
+	public void setLabel(List<String> label) {
+		this.labels = label;
+	}
+
+	@Override
+	public Map<String, String> getDescription() {
+		return this.descriptions;
+	}
+
+	@Override
+	public void setDescription(Map<String, String> descr) {
+		this.descriptions = descr;
+	}
+
+	@Override
+	public List<TermEntity> getTopicTerms() {
+		return this.terms;
+	}
+
+	@Override
+	public void setTopicTerms(List<TermEntity> terms) {
+		this.terms = terms;
+	}
+
+	@Override
+	public List<TermEntity> getTopicKeywords() {
+		return this.keywords;
+	}
+
+	@Override
+	public void setTopicKeywords(List<TermEntity> keywords) {
+		this.keywords = keywords;
+	}
+
+	@Override
+	public TopicModel getTopicModel() {
+		return this.model;
+	}
+
+	@Override
+	public void setTopicModel(TopicModel model) {
+		this.model = model;
+
+	}
+
+	@Override
+	public Date getCreatedDate() {
+		return this.created;
+	}
+
+	@Override
+	public Date getModifiedDate() {
+		return this.modified;
+	}
+
+	@Override
+	public void setCreatedDate(Date date) {
+		this.created = date;
+	}
+
+	@Override
+	public void setModifiedDate(Date date) {
+		this.modified = date;
+	}
+	
+	@Override
+	public String getId() {
+		return _id;
+	}
+
+	@Override
+	public String toJSON() throws JsonProcessingException {
```

Comment:
the serialization must not be part of the data model, the serializer bean should be used for serialization of all entities in the Controllers (not in Services)

Reply:
these applies to all data model classes